File: 67.Add-Binary.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function (a, b) {
  let result = "";
  let cary = 0;
  let i = a.length - 1;
  let j = b.length - 1;

  while (i >= 0 || j >= 0 || cary) {
    const sum = cary + (a[i] ? +a[i] : 0) + (b[j] ? +b[j] : 0);
    result = (sum % 2) + result;
    cary = Math.floor(sum / 2);

    i -= 1;
    j -= 1;
  }

  return result;
};

File: python/67.Add-Binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        result = ""
        i = len(a) - 1
        j = len(b) - 1
        cary = 0

        while i >= 0 or j >= 0 or cary:
            total = cary + (int(a[i]) if i >= 0 else 0) + (int(b[j]) if j >= 0 else 0)

            result = str(total % 2) + result
            cary = total // 2

            i -= 1
            j -= 1

        return result